### PR TITLE
Django 1.11 support

### PR DESCRIPTION
**Pull request by rtpg**
This tries to pull in various 1.11 fixes into one place, as well as test 1.11 properly 

(This also reduces the scope of what the project tests to supported versions of Django)

### CODE REVIEW

File: floppyforms/widgets.py [hunk 1]
```diff
@@ -78,12 +96,14 @@ def _format_value(self, value):
 
     def get_context(self, name, value, attrs=None):
         context = {
+            'widget': self,
```
Comment (rtpg) [inline floppyforms/widgets.py]: Django's default form templates expect `widget`

File: tests/settings.py [hunk 2]
```diff
@@ -23,10 +27,19 @@
...
 
+if django.VERSION < (1, 11):
+    template_directories = []
+else:
+    template_directories = [
+        os.path.join(
+            os.path.dirname(django.__file__),
+            "forms/templates/"
+        )
+    ]
```
Comment (rtpg) [inline tests/settings.py]: this code block is necessary for finding django 1.11's form templates. Though for the most part they aren't used, a couple widgets rely on their base widgets for rendering by default still.

**Comment by rtpg** [comment]
I've finally gotten this to pass tests in all supported Django versions. 

For the most part there are no major changes here, except that template loaders in Django 1.11 will require adding the `django/forms/templates` directory to their possible template folders, since some code paths end up holding onto Django widgets that are configured to use specific kinds of template names.

**Comment by mrjmad** [comment]
:+1:

**Comment by hsmett** [comment]
Hello @gregmuellegger !
This pull request seems to sucessfully bring django 1.11 support to the project,  can you please take a look?
Thank you! :D

**Comment by dsteinberger** [comment]
hi @gregmuellegger 

It's possible to merge this PR please ?

Thanks

**Comment by jonasmcarson** [comment]
hi @gregmuellegger

Would really appreciate if you could look into merging this PR

Thanks

**Comment by psyferre** [comment]
This repo appears to be abandoned... had to remove from all my projects so I could upgrade past Django 1.11.3

**Pull request merged**